### PR TITLE
Enhance `filters.gpstimeconvert` to make it streamable

### PR DESCRIPTION
**Pull request by gui2dev**
Implement `processOne` to make `Filters.gpstimeconvert` streamable.

### CODE REVIEW

File: filters/GpsTimeConvert.cpp [hunk 1]
```diff
@@ -85,14 +83,18 @@ void GpsTimeConvert::testTimeType(std::string& type)
...
 {
+    m_first = true;
+    m_lastTime = std::numeric_limits<double>::lowest();
+    m_numSeconds = std::numeric_limits<int>::lowest();
     if (!m_conversion.empty() && (m_inTime.empty()) && (m_outTime.empty()))
     {
         m_conversion = Utils::tolower(m_conversion);
-        std::vector<std::string> s = Utils::split(m_conversion,'2');
+        std::vector<std::string> s = Utils::split(m_conversion, '2');
         m_inTime = s[0];
```
Comment (abellgithub) [inline filters/GpsTimeConvert.cpp:93]: This should check that the vector contains (at least) two items.

**Comment by hobu** [comment]
This needs a `run` method too. See ReprojectionFilter for inspiration.

**Comment by hobu** [comment]
> This needs a `run` method too. See ReprojectionFilter for inspiration.

@abellgithub we often forget to add the `run` method. I wonder if we shouldn't provide a default implementation.

File: filters/GpsTimeConvert.cpp [hunk 2]
```diff
@@ -193,225 +210,174 @@ int GpsTimeConvert::dayStartGpsSeconds(std::tm date)
...
 
     return duration.count();
 }
 
-void GpsTimeConvert::unwrapWeekSeconds(PointView& view)
+void GpsTimeConvert::unwrapWeekSeconds(PointRef& point)
 {
     // any decrease in time is interpreted as a week rollover
-    for (PointId i = 0; i < (view.size()-1); ++i)
+    while (point.getFieldAs<double>(Dimension::Id::GpsTime) < m_lastTime)
```
Comment (gui2dev) [inline filters/GpsTimeConvert.cpp:222]: @hobu, @abellgithub we realized that test is somehow too strict. For some point cloud, you might have little decrease of `GpsTime` (namely when multiple returns).
We introduced another parameter called `wrapped_tolerance` that would update the test like this :
`point.getFieldAs<double>(Dimension::Id::GpsTime) + m_wrappedTolerance < m_lastTime`

Is that ok if I put this modification in this PR ?

Comment (hobu) [reply filters/GpsTimeConvert.cpp:222]: > Is that ok if I put this modification in this PR ?

👍

**Comment by hobu** [comment]
Ready to merge @gui2dev ?

**Comment by gui2dev** [comment]
> Ready to merge @gui2dev ?

Absolutely !

**Pull request merged**